internal/controller/ingress/service_controller.go

```diff
@@ -108,6 +124,34 @@ func (r *ServiceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		}
 	}
 
+	// Index the services by the module set(s) they reference
+	err := mgr.GetFieldIndexer().IndexField(context.Background(), &corev1.Service{}, ModuleSetPath, func(obj client.Object) []string {
```

Comment:
this is new to me, is this just a more efficient way than to get all the objects or something?

Reply:
This indexes the services by the module sets the use. This allows us to efficiently answer the question: "which services need to be reconciled when a module set changes" and then trigger the reconciles for those services. For modulesets, of which you can have multiple in the annotation, this comment in the FieldIndexer is important:

https://github.com/kubernetes-sigs/controller-runtime/blob/6f1c3d819175d2c2f72865c6e544b0f78dc8c280/pkg/client/interfaces.go#L196-L202